File: projects/designpatterns/1. Strategy Pattern/strategy_pattern_Duck_cpp/MiniDuckSimulator.cpp
#include <iostream>
#include "MallardDuck.h"
#include "RubberDuck.h"
#include "FlyRocketPowered.h"

using namespace std;

int main()
{
	Duck *mallard = new MallardDuck();
	mallard->performFly();

	Duck *rubber = new RubberDuck();
	rubber->performFly();

	rubber->setFlyBehavior(new FlyRocketPowered());
	rubber->performFly();

	mallard->performQuack();
	rubber->performQuack();

	delete mallard;
	delete rubber;

	return 0;
}

File: projects/designpatterns/1. Strategy Pattern/strategy_pattern_Duck_java/MiniDuckSimulator.java
public class MiniDuckSimulator {

	public static void main(String[] args) {

		Duck mallard = new MallardDuck();
		mallard.performFly();

		Duck rubber = new RubberDuck();
		rubber.performFly();

		rubber.setFlyBehavior(new FlyRocketPowered());
		rubber.performFly();

		mallard.performQuack();
		rubber.performQuack();
	}
}